From a1175bdb3e69bd363616b323c97f79c1853283ec Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Thu, 3 Mar 2011 16:55:55 -0300
Subject: [PATCH 08/12] Revert "qdev: Don't leak string property value on hot unplug"

RH-Reverts: abf92e48152ebadb2ec8caf34199b4a7ac83f466
RH-Reverts-patchwork-id: 18383
Bugzilla-related: 677170
Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 hw/qdev-properties.c |    6 ------
 hw/qdev.c            |    6 ------
 hw/qdev.h            |    1 -
 3 files changed, 0 insertions(+), 13 deletions(-)

diff --git a/hw/qdev-properties.c b/hw/qdev-properties.c
index 169b9a2..64d83ce 100644
--- a/hw/qdev-properties.c
+++ b/hw/qdev-properties.c
@@ -280,11 +280,6 @@ static int parse_string(DeviceState *dev, Property *prop, const char *str)
     return 0;
 }
 
-static void free_string(DeviceState *dev, Property *prop)
-{
-    qemu_free(*(char **)qdev_get_prop_ptr(dev, prop));
-}
-
 static int print_string(DeviceState *dev, Property *prop, char *dest, size_t len)
 {
     char **ptr = qdev_get_prop_ptr(dev, prop);
@@ -299,7 +294,6 @@ PropertyInfo qdev_prop_string = {
     .size  = sizeof(char*),
     .parse = parse_string,
     .print = print_string,
-    .free  = free_string,
 };
 
 /* --- drive --- */
diff --git a/hw/qdev.c b/hw/qdev.c
index 023f423..7a7e344 100644
--- a/hw/qdev.c
+++ b/hw/qdev.c
@@ -325,7 +325,6 @@ void qdev_init_nofail(DeviceState *dev)
 void qdev_free(DeviceState *dev)
 {
     BusState *bus;
-    Property *prop;
 
     if (dev->state == DEV_STATE_INITIALIZED) {
         while (dev->num_child_bus) {
@@ -341,11 +340,6 @@ void qdev_free(DeviceState *dev)
     }
     qemu_unregister_reset(qdev_reset, dev);
     QLIST_REMOVE(dev, sibling);
-    for (prop = dev->info->props; prop && prop->name; prop++) {
-        if (prop->info->free) {
-            prop->info->free(dev, prop);
-        }
-    }
     qemu_free(dev);
 }
 
diff --git a/hw/qdev.h b/hw/qdev.h
index 2a16949..f9791f1 100644
--- a/hw/qdev.h
+++ b/hw/qdev.h
@@ -99,7 +99,6 @@ struct PropertyInfo {
     enum PropertyType type;
     int (*parse)(DeviceState *dev, Property *prop, const char *str);
     int (*print)(DeviceState *dev, Property *prop, char *dest, size_t len);
-    void (*free)(DeviceState *dev, Property *prop);
 };
 
 typedef struct GlobalProperty {
-- 
1.7.4.rc1.16.gd2f15e

